src/goto-instrument/contracts/contracts.cpp

```diff
@@ -86,6 +92,15 @@ void code_contractst::check_apply_loop_contracts(
     invariant = conjunction(invariant.operands());
   }
 
+  // Vector representing a (possibly multidimensional) decreases clause
+  const auto decreases_clause_vector = decreases_clause.operands();
+  const size_t size_decreases_clasue = decreases_clause_vector.size();
```

Comment:
You don't need to store the size on a separate variable, you can just call `size()`. Even better, vectors are traversed using iterators.

Reply:
Oh, yes, you're right. I should use an iterator.

Reply:
Actually, I still need the index variable `i` because I traverse multiple vectors (of the same length) simultaneously. So as you suggested, I will access the size via `size()` instead of storing it in a designated variable.